src/fides/api/ops/api/v1/endpoints/privacy_preference_endpoints.py

```diff
@@ -241,21 +295,117 @@ def save_privacy_preferences(
             detail=privacy_request_results.failed[0].message,
         )
 
-    consent_request.privacy_request_id = privacy_request_results.succeeded[0].id
-    consent_request.save(db=db)
-
+    if consent_request:
+        # If we have a verified user identity, go ahead and update the associated ConsentRequest for record keeping
+        consent_request.privacy_request_id = privacy_request_results.succeeded[0].id
+        consent_request.save(db=db)
     return upserted_current_preferences
 
 
+def verify_address(request: Request) -> None:
+    """Verify request is coming from approved address"""
+    origin = request.headers.get("Origin") or request.headers.get("Referer")
+
+    if origin:
+        parsed_url = urlparse(origin)
+        if (
+            parsed_url.scheme + "://" + parsed_url.netloc
+            not in CONFIG.security.cors_origins
+        ):
+            raise HTTPException(
+                status_code=HTTP_403_FORBIDDEN,
+                detail="Can't save privacy preferences from non-approved addresses",
+            )
+
+
+@router.get(
+    PRIVACY_PREFERENCES,
+    status_code=HTTP_200_OK,
+    response_model=Page[CurrentPrivacyPreferenceSchema],
+)
+def get_privacy_preferences_by_device_id(
+    *,
+    request: Request,
+    fides_user_device_id: str,
+    db: Session = Depends(get_db),
+    params: Params = Depends(),
+) -> AbstractPage[CurrentPrivacyPreference]:
+    """Saves privacy preferences with respect to a fides user device id.
```

Comment:
I think this docstr needs to be updated to match this GET endpoint

Reply:
ah thanks copy paste